### PR TITLE
[Bug fix] Input is changed when rerun the node after run the overall flow

### PR DESCRIPTION
**Pull request by lalala123123**
# Description
node test supports dict as input.
inputs.jsonl generated by extension:
```
{
    "get_dict_val.output.value": {"key": "value"},
}
```
pf flow test --node xxx --inputs inputs.jsonl
Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_sdk/_submitter/test_submitter.py [hunk 1]
```diff
@@ -116,9 +116,12 @@ def resolve_data(
                     else:
                         missing_inputs.append(name)
                         continue
-                    dependency_nodes_outputs[value.value] = (
-                        {value.property: dependency_input} if value.property else dependency_input
-                    )
+                    if value.property:
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_submitter/test_submitter.py:119]: please add more PR description on this change.

File: src/promptflow/promptflow/_sdk/_submitter/test_submitter.py [hunk 2]
```diff
@@ -116,9 +116,12 @@ def resolve_data(
                     else:
                         missing_inputs.append(name)
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_submitter/test_submitter.py:117]: pls add a changelog

File: src/promptflow/promptflow/_sdk/_submitter/test_submitter.py (hunk 1 above)
Comment (lalala123123) [reply src/promptflow/promptflow/_sdk/_submitter/test_submitter.py:119]: update PR description

File: src/promptflow/promptflow/_sdk/_submitter/test_submitter.py (hunk 2 above)
Comment (lalala123123) [reply src/promptflow/promptflow/_sdk/_submitter/test_submitter.py:117]: update changelog

**Review by wangchao1230 — DISMISSED** [review]

**Review by zhengfeiwang — APPROVED** [review]

**Review by zhengfeiwang — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**